path/posix/path.mbt

```diff
@@ -0,0 +1,181 @@
+// Copyright 2025 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+///|
+/// Returns the last path component of the given path.
+///
+/// property: `@path.join(@path.dirname(path), @path.basename(path)) == path`
+/// 
+/// Warning: basename function not respect this POSIX standard, but consistent with Python `os.path` module.
+/// 
+pub fn basename(path : StringView) -> StringView {
+  match path.rev_find("/") {
+    None => path.to_string()
+    Some(idx) if idx >= path.length() => ""
+    Some(idx) => try! path[idx + 1:].to_string()
+  }
+}
+
+///|
+/// Returns path without its last path component and trailing slashes.
+/// 
+/// property: `@path.join(@path.dirname(path), @path.basename(path)) == path`
+/// 
+/// Warning: basename function not respect this POSIX standard, but consistent with Python `os.path` module.
+/// 
+pub fn dirname(path : StringView) -> StringView {
+  let last_slash_pos = path.rev_find("/")
+  match last_slash_pos {
+    None => ""
+    Some(idx) => {
+      let dirname_with_trailing_slashes = try! path[0:idx]
+      let dirname_trim_end = dirname_with_trailing_slashes.trim_end(
+        char_set="/",
+      )
+      if dirname_trim_end.is_empty() {
+        dirname_with_trailing_slashes.to_string()
+      } else {
+        dirname_trim_end.to_string()
+      }
+    }
+  }
+}
+
+///|
+/// Returns the string of the basename after the last `.` delimiter.
+pub fn extname(path : StringView) -> StringView {
+  let basename = basename(path)
+  match basename.rev_find(".") {
+    None => ""
+    Some(idx) => try! basename[idx:].to_string()
+  }
+}
+
+///|
+/// Returns whether the given path is absolute.
+/// 
+/// edge cases: when path is empty, return false
+pub fn is_absolute(path : StringView) -> Bool {
+  match path {
+    ['/', ..] => true
+    _ => false
+  }
+}
+
+///|
+/// 1. string concatenation with OS platform specific path separator
+/// 2. if rhs is absolute path, return rhs
+/// 
+/// edge cases:
+///  1. when lhs and rhs both are empty, return empty string
+///  2. ignore empty string
+pub fn join(lhs : StringView, rhs : StringView) -> StringView {
```

Comment:
Rationale for returning `StringView` here? All branches calls `.to_string()` and `String` is more appropriate?